BZ1210240: Fixes/cleanup for "Basic CLI Operations"

adellape opened the pull request:
Also:

- Moved the "Object Types" table from `get_started_cli.adoc` to `basic_cli_operation.adoc` where it has more context.
- Added "What's Next?" to the end of `get_started_cli.adoc` to link to `basic_cli_operations.adoc` and `new_app.adoc`.
- Fixed broken links in these topics due to restructure.

adellape commented:
@bfallonf PTAL (added you as QA for [BZ#1210240](https://bugzilla.redhat.com/show_bug.cgi?id=1210240))

bfallonf commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -180,28 +272,36 @@ Use the `osc rollback` command to revert part of an application back to a previo
 .^|`-t, --template`
-|Template string or path to template file to use when -o=template or -o=templatefile.
+|Template string or path to template file to use when ``-o=template` or `-o=templatefile`.
 |===
```
@adellape Is there an extra ' in the line above? Eg: ``-o=template'

CowboysFan replied:
There's no reason to have the command options spread out across two lines, they should be on a single line. Unless it's just GitHub formatting, you should increase the width of the first column to ensure that.

adellape replied:
@bfallonf Good catch.

@CowboysFan It's just GH; looks fine in the build.

bfallonf commented on cli_reference/get_started_cli.adoc:
```diff
@@ -24,15 +30,20 @@ Logged into 'https://my-openshift-server.com' as 'myuser'.
+- The command prompted for the OpenShift server URL and the login credentials (a user name and password), established a session in the server, and received a session token.
+- One OpenShift project the user has access to was selected and set as default.
+- All information gathered were saved in a config file located in *_~/.config/openshift/.config_*.
 
```
@adellape Should this be "All information gathered was saved..."?

Or maybe "The gathered information was saved..."?

CowboysFan replied:
I think this would be a great place to use callouts to describe each of these steps, rather than just a bullet list.

adellape replied:
Both comments addressed.

bfallonf commented on cli_reference/get_started_cli.adoc:
```diff
@@ -52,25 +63,36 @@ users:
+====
+
+From now on, every subsequent command will make use of the server, session token
+and project referenced in that config file. Notice what happens if you run `osc
```
@adellape Little nitpick, but comma after 'token" here?

adellape replied:
oxford.comma++

Fixed.

bfallonf commented on cli_reference/get_started_cli.adoc:
```diff
@@ -52,25 +63,36 @@ users:
 
-Configuration file permanently stores `osc` options; it contains a series of authentication mechanisms and server connection information associated with nicknames. As described in the previous section, the `osc login` command is capable of creating and managing configuration files automatically.
+Configuration file permanently stores `osc` options; it contains a series of
+authentication mechanisms and server connection information associated with
```
@adellape Should this start with "Configuration files permanently store `osc` options"? Or maybe "A configuration file permanently stores `osc` options"?

adellape replied:
Fixed.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -10,34 +10,56 @@
-This topic provides information on some general CLI operations and their syntax.
+This topic provides information on some general CLI operations and their syntax,
+and assumes you have gone through the link:get_started_cli.html[basic setup and
+login] for the CLI.
```
[Suggestion]
This topic provides information on some general CLI operations and their syntax. You must link:get_started_cli.html[configure and set up] the CLI before you can perform these operations.

Generally we try not to assume and just state what you must already have done before you can do anything in this topic, if that's the case.

adellape replied:
Fixed.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -1,4 +1,4 @@
-= CLI Operations
+= Basic CLI Operations
```
I don't think it makes sense to title the whole topic Basic CLI Operations while we have sections that are titled Common CLI Operations, Project Operations, etc.

It may be better to change the name in the nav bar to match the name of this topic, CLI Operations.

adellape replied:
Agreed. The naming was inconsistent (between nav title, topic title, headings, file name) and kinda strange. I think we need to define the scope of this topic: Touch briefly on various common types of operations in each subsection (projects, builds, deployments, etc) and link to the dedicated topics for detailed info/usage? Be a comprehensive reference for _all_ CLI operations? 

For now, I'll synced up the topic title and nav bar to "CLI Operations". Leaving the file name as `basic_cli_operations` for now (don't want to break any more links), but that may need to be synced up in the future as well if something better overall is landed on.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -10,34 +10,56 @@
+- The `_<object_name_or_id>_` of the specified `_<object_type>_`.
 
+For example, the `osc get` operation returns a complete list of services that are currently defined:
+
```
[Suggestion]
The following example shows the `osc get` operation return a complete list of services that are currently defined.

[Example]
Wrap the example in an example block.

adellape replied:
It already is in an example block... ?

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -10,34 +10,56 @@
+No events.
 ----
+====
 
```
Again, wrap the example in an Example block.

adellape replied:
Already is.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -72,24 +86,60 @@ The following table describes the common `osc` operations and their general synt
+====
+
+This creates a new project called `test` that appears in the Management Console as "Openshift 3 Sample", with `test-admin` as the project admin.
 
```
Again, use the Example block.

The following example creates a new project called `test` that appears in the Management Console as "Openshift 3 Sample", with `test-admin` as the project admin.

[Example]
screen output

adellape replied:
Already is in an example block.

Updated lead-in sentence.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -136,32 +190,70 @@ One of the fundamental capabilities of OpenShift is the ability to build applica
 
-The syntax to describe a deployment configuration in human-readable form is:
+To describe a deployment configuration in human-readable form:
 
```
The sentence "To describe a deployment configuration in human-readable form:" isn't very clear as to what you're doing. It's better to state that you use this syntax to describe a deployment config.

[Suggestion]
Use the following syntax to describe a deployment configuration in human-readable form:

adellape replied:
Updated.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -136,32 +190,70 @@ One of the fundamental capabilities of OpenShift is the ability to build applica
+	Pods Status:	1 Running / 0 Waiting / 0 Succeeded / 0 Failed
+----
+====
 
```
Again, use the Example block.

The following example.....

[Example]
screen output

adellape replied:
Already is in an example block.

Updated lead-in sentence.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -136,32 +190,70 @@ One of the fundamental capabilities of OpenShift is the ability to build applica
+environment variables and volumes are included in rollbacks, so there are a
+variety of considerations to make when deciding whether a rollback is viable,
+such as:
 
```
[Suggestion]
Rollbacks revert an application back to a previous deployment, and they include environment variable and volumes. Therefore, when you decide whether a rollback is viable or not, among other things, consider the following:

adellape replied:
Updated.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -136,32 +190,70 @@ One of the fundamental capabilities of OpenShift is the ability to build applica
-Use the `osc rollback` command to revert part of an application back to a previous deployment. The syntax for `rollback` is:
+By default, only the pod and container configuration are changed and scaling or
+trigger settings are left as-is.
+
```
By default, only the pod and container configuration are changed and scaling or trigger settings are left as-is. ==>> During a rollback, only the configuration of pods and containers is changed by default, while the scaling or trigger settings remain unchanged.

Note:
Part of our goal is to try and eliminate community tone and raise the level of professionalism in our docs.

adellape replied:
Updated.

CowboysFan commented on cli_reference/basic_cli_operations.adoc:
```diff
@@ -136,32 +190,70 @@ One of the fundamental capabilities of OpenShift is the ability to build applica
+
+Use the `osc rollback` command to revert part of an application back to a
+previous deployment:
 
```
[Suggestion]
The `-d`, or `--dry run` option shows the configuration of the updated deployment in an easy to read format without actually executing the rollback. This allows you to inspect the output before actually proceeding with the rollback.

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -10,13 +10,19 @@
+establishing a session against an OpenShift server with the provided
+credentials. It then *automatically saves* a configuration file that is used by
+every subsequent command.
 
```
Why is some of the text unnecessarily in *bold*? 

[Suggestion]
The `osc login` command is the best way to set up and configure the OpenShift CLI, and it serves as the entry point for most users. The interactive flow helps you establish a session to an OpenShift server with the provided credentials. The configuration is automatically saved and is then used for every subsequent command.

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -24,15 +30,20 @@ Logged into 'https://my-openshift-server.com' as 'myuser'.
 Welcome to OpenShift v3! Use 'osc --help' for a list of commands available.
 ----
+====
 
```
Please wrap this in an Example block and provide an intro sentence to the example, as per our guidelines.

adellape replied:
Already in an example block.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -24,15 +30,20 @@ Logged into 'https://my-openshift-server.com' as 'myuser'.
-Here's how the content of the config file just created in `~/.config/openshift/.config` looks like:
+The content of the configuration file created in
+*_~/.config/openshift/.config_* looks like:
 
```
This is not the recommended practice for our docs, and this sentence should be reworded, to something like:

[Suggestion]
The following example shows a sample configuration file that is created in *_~/.config/openshift/.config_*.

[Example]
screen output

adellape replied:
Moved this content to the "CLI Configuration Files" section.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -52,25 +63,36 @@ users:
+From now on, every subsequent command will make use of the server, session token
+and project referenced in that config file. Notice what happens if you run `osc
+login` again:
 
```
This needs to be reworded and brought up to CCS standards. 

[Suggestion]
When you have completed the CLI configuration, every subsequent command uses the configuration file for the server, session token, and project information.

The following example shows the output of the `osc login` command after the CLI has been configured.
[Example]
screen output

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -52,25 +63,36 @@ users:
+authentication mechanisms and server connection information associated with
+nicknames. As described in the previous section, the `osc login` command is
+capable of creating and managing configuration files automatically.
 
```
[Suggestion]
The configuration file permanently stores `osc` options and contains a series of authentication mechanisms and server connection information associated with nicknames. As described in the previous section, the `osc login` command can automatically create and manage configuration files.

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -93,9 +115,15 @@ users:
+easily between them. Multiple configuration files are allowed; they are loaded
+at runtime and merged together along with override options specified from the
+command line.
 
```
[Suggestion]
Configuration files allow you to set up multiple users, servers, and namespaces to use with an OpenShift environment. You can use multiple configuration files that are loaded at runtime and combined with the override options specified from the CLI.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -93,9 +115,15 @@ users:
+There are a few ways to provide `osc` a configuration file. Basically when
+loading configurations the following priority rules for configuration file
+loading are observed (in this exact order):
 
```
[Suggestion]
There are multiple methods that you can use to create a configuration file to be used with `osc`. The following rules are  observed in order when a configuration file is loaded:

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -105,13 +133,23 @@ There are a few ways to provide `osc` a configuration file. Basically when loadi
+contain multiple
+link:../architecture/core_objects/openshift_model.html[components] that compose
+your logical "link:../whats_new/applications.html[application]".
 
```
[Suggestion]
A project in OpenShift is the package that contains multiple components to make up a logical application.

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -105,13 +133,23 @@ There are a few ways to provide `osc` a configuration file. Basically when loadi
+link:../dev_guide/projects.html[project]. When running `osc login`, a project is
+set as default to be used by every subsequent command, and if you have access to
+more than one project, one of them is picked:
 
```
[Suggestion]
When you run the `osc login` command to configure the CLI, a default project is set to be used with subsequent commands. If you have access to multiple projects then one is selected, as shown in the following example.

[Example]
screen output

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -135,9 +178,16 @@ Now using project 'project03'.
 $ osc project
 Using project 'project03'.
 ----
```
There's a lot going on in this paragraph, and we generally show the commands in the sidebar block, not as part of the intro text. I think we should separate this into the different tasks. For example:

Use the following command to view the project you are currently using:
`osc project`

Switch to a particular project by specifying the name of the project:
`osc project _<project>_`

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -135,9 +178,16 @@ Now using project 'project03'.
+local git repository, the `osc status` command shows a high level overview of
+the project currently in use, with its components and their relationships. This
+command takes no argument:
 
```
Why do we have this statement - "The same way `git status` provides....."? I don't see any relevance of this statement in this context.

[Suggestion]
The `osc status` command shows a high level overview of the project currently in use, with its components and their relationships, as shown in the following example.

[Example]
screen output

adellape replied:
Updated.

CowboysFan commented on cli_reference/get_started_cli.adoc:
```diff
@@ -153,27 +203,12 @@ service frontend-test (172.30.17.236:5432 -> 8080)
\ No newline at end of file
+After you have link:get_started_cli.html#basic-setup-and-login[logged in], check
+out some link:basic_cli_operations.html[basic CLI operations], or get straight
+into link:../dev_guide/new_app.html[creating a new application].
```
This is not appropriate wording for Enterprise docs:

"Check out some....., or get straight into....."

Please modify this.

adellape replied:
Updated.

CowboysFan commented:
@adellape I'm assuming you were done since you asked for a peer review, so I made some inline comments. Thanks.

adellape commented:
@CowboysFan Thanks. This PR was initially just about addressing the referenced BZ and a couple tangential fixes (hadn't planned on a full passthrough of the existing content). But since you've gone through and done the full review, I've gone in and addressed the content per your comments/suggestions and updated the PR.

Will merge after build passes.

CowboysFan commented:
@adellape Yeah, that's why I think I asked you in the email. I wasn't sure so I went through the whole thing. Thanks.
